### PR TITLE
Pr 2437 fixes

### PR DESCRIPTION
**Pull request by thunder2**

### CODE REVIEW

File: libretroshare/src/pqi/pqifdbin.cc [hunk 1]
```diff
@@ -53,8 +54,11 @@ void RsFdBinInterface::setSocket(int s)
 
 #else
     // On windows, there is no way to determine whether a socket is blocking or not, so we set it to non blocking whatsoever.
-    unsigned long int on = 1;
-    ioctlsocket(s, FIONBIO, &on);
+    if (mIsSocket) {
```
Comment (csoler) [inline libretroshare/src/pqi/pqifdbin.cc:57]: I think we should only set the fd to non blocking if it is a socket. Otherwise, it's meaningless. In posix, this has no impact. On windows I don't know.

File: libretroshare/src/tor/TorControl.cpp [hunk 2]
```diff
@@ -68,6 +68,11 @@ TorControl::TorControl()
     : mControlPort(0),mSocksPort(0),mStatus(NotConnected), mTorStatus(TorOffline),mHasOwnership(false)
 {
     mSocket = new TorControlSocket(this);
+    mControlPort = 0;
```
Comment (csoler) [inline libretroshare/src/tor/TorControl.cpp:71]: Looks like we both fixed that at the same time ;-)

File: libretroshare/src/util/rsfile.cc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+#ifdef WINDOWS_SYS
+#include <wtypes.h>
+#include <io.h>
+#include <namedpipeapi.h>
+#include <errno.h>
+#else
+#include <fcntl.h>
+#endif
+
+int RsFileUtil::set_fd_nonblock(int fd)
```
Comment (csoler) [inline libretroshare/src/util/rsfile.cc:34]: why did you need to create this function? unix_fcntl_nonblock() does this already (for sockets), but it should only be called for sockets anyway.

File: libretroshare/src/pqi/pqifdbin.cc (hunk 1 above)
Comment (thunder2) [reply libretroshare/src/pqi/pqifdbin.cc:57]: If the file descriptor is not set to non block, read will wait until data is available on Windows. It is not necessary when the caller sets the file descriptor to non block with RsFileUtil::set_fd_nonblock like in TorProcess::start. I have only added the right function for file descriptor, because unix_fcntl_nonblock shows an error in log for a file descriptor.

File: libretroshare/src/util/rsfile.cc (hunk 3 above)
Comment (thunder2) [reply libretroshare/src/util/rsfile.cc:34]: I thought unix_fcntl_nonblock can be used for file descriptors and sockets on Linux. A call to unix_fcntl_nonblock was already there in TorProcess::start for stdout and stderr.
On Windows there are two different functions to set a fd to non blocking. The Windows specific part in unix_fcntl_nonblock can set a socket to non blocking and the Windows specific part in RsFileUtil::set_fd_nonblock can set a file descriptor like stdout and stderr to non blocking.
The Linux specific part is the same in both functions. Maybe one function can call the other to not have the same code twice.

File: libretroshare/src/tor/TorControl.cpp (hunk 2 above)
Comment (thunder2) [reply libretroshare/src/tor/TorControl.cpp:71]: Yes, there were some strange behaviors without the initialization.
